app/src/main/java/org/p2p/wallet/history/HistoryModule.kt

```diff
@@ -26,88 +26,31 @@ import org.p2p.wallet.history.ui.token.TokenHistoryContract
+        factoryOf(::HistoryInteractor)
+        factoryOf(::HistoryItemMapper)
 
-        factory { (token: Token.Active) ->
-            TokenHistoryPresenter(
-                token = token,
-                historyInteractor = get(),
-                receiveAnalytics = get(),
-                swapAnalytics = get(),
-                analyticsInteractor = get(),
-                sendAnalytics = get(),
-                renBtcInteractor = get(),
-                tokenInteractor = get(),
-                sellInteractor = get()
-            )
-        } bind TokenHistoryContract.Presenter::class
-        factory { (state: TransactionDetailsLaunchState) ->
-            TransactionDetailsPresenter(
-                resourcesProvider = get(),
-                state = state,
-                userLocalRepository = get(),
-                historyInteractor = get()
-            )
-        } bind TransactionDetailsContract.Presenter::class
-        factory { (state: TransactionDetailsLaunchState) ->
-            HistoryTransactionDetailsBottomSheetPresenter(
-                state = state,
-                historyInteractor = get(),
-                usernameInteractor = get()
-            )
-        } bind HistoryTransactionDetailsContract.Presenter::class
+        factoryOf(::HistoryPresenter) bind HistoryContract.Presenter::class
+        factoryOf(::TokenHistoryPresenter) bind TokenHistoryContract.Presenter::class
+        factoryOf(::TransactionDetailsPresenter) bind TransactionDetailsContract.Presenter::class
+        factoryOf(::HistoryTransactionDetailsBottomSheetPresenter) {
+            bind<HistoryTransactionDetailsContract.Presenter>()
+        }
     }
 
     private fun Module.dataLayer() {
-        factory { TransactionDetailsEntityMapper(get()) }
+        factoryOf(::TransactionDetailsEntityMapper)
         singleOf(::TransactionDetailsDatabaseRepository) bind TransactionDetailsLocalRepository::class
-
-        single {
-            val api = get<Retrofit>(named(RpcModule.RPC_RETROFIT_QUALIFIER))
-                .create(RpcHistoryApi::class.java)
-
-            TransactionDetailsRpcRepository(
-                rpcApi = api,
-                transactionParsingContext = get()
-            )
-        } bind TransactionDetailsRemoteRepository::class
-
-        factory {
-            HistoryPresenter(
-                historyInteractor = get(),
-                renBtcInteractor = get(),
-                receiveAnalytics = get(),
-                swapAnalytics = get(),
-                analyticsInteractor = get(),
-                environmentManager = get(),
-                sendAnalytics = get(),
-                sellInteractor = get()
-            )
-        } bind HistoryContract.Presenter::class
+        single { get<Retrofit>(named(RpcModule.RPC_RETROFIT_QUALIFIER)).create(RpcHistoryApi::class.java) }
+        singleOf(::TransactionDetailsRpcRepository) bind TransactionDetailsRemoteRepository::class
```

Comment:
maybe you'd better to create an api within `single` block of repository? 
You would avoid adding one more dependency in the graph

Reply:
well, okay, will fix it in another pr